#!/usr/bin/env python
PACKAGE = "drive_ros_image_recognition"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("sobelThreshold", int_t, 0, "Sobel threshold", 50, 0, 200)
gen.add("minLineWidthMul", double_t, 0, "minLineWidthMul",  0.3, 0.0, 1.0)
gen.add("maxLineWidthMul", double_t, 0, "maxLineWidthMul", 3.0, 0.0, 5.0)
gen.add("lineWidth", double_t, 0, "Maximum line width considered for stopping lines", 0.04, 0.0, 0.1)
gen.add("offsetSide", double_t, 0, "Distance moved left/right for additional checks", 0.1, 0.0, 0.5)
gen.add("offsetAlong", double_t, 0, "Length to which search lines will be normalized", 0.2, 0.0, 0.5)
gen.add("oppositeLineStart", double_t, 0, "Distance offset to opposite crossing line start", 0.6, 0.0, 0.2)
gen.add("oppositeLineEnd", double_t, 0, "Distance offset to opposite crossing line end", 1.2, 0.0, 2.0)
gen.add("oppositeLineOffset", double_t, 0, "Offset for opposite search line search", 0.85, 0.1, 1.0)
gen.add("crossingCheckOffset", double_t, 0, "Offset to the left for crossing line check", -0.4, -0.6, -0.1)

exit(gen.generate(PACKAGE, "drive_ros_image_recognition", "CrossingDetection"))

